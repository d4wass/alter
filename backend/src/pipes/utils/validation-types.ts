import { ValidationError } from '@nestjs/common';

export type ConvertedValidationError = {
  property: string;
  constraints: any;
  children?: ValidationError[];
};

export type ConstructorTypes =
  | StringConstructor
  | BooleanConstructor
  | NumberConstructor
  | ArrayConstructor
  | ObjectConstructor;
