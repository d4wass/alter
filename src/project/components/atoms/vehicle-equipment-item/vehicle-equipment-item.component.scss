.remove-btn {
  background: none;
  border: none;
  cursor: pointer;
}

.equipment-item {
  display: flex;
  justify-content: space-between;
  align-items: center;
}
