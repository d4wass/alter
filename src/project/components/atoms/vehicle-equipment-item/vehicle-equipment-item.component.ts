import { Component, EventEmitter, Input, Output } from '@angular/core';

@Component({
    selector: 'app-vehicle-equipment-item',
    template: `
    <mat-list-item class="equipment-item">
      <div matListItemTitle data-test="title">{{ equipment }}</div>
      <button (click)="closeClick($event)" class="remove-btn" data-test="remove-btn">
        <mat-icon matListItemIcon>close</mat-icon>
      </button>
    </mat-list-item>
  `,
    styleUrls: ['./vehicle-equipment-item.component.scss'],
    standalone: false
})
export class VehicleEquipmentItemComponent {
  @Input() equipment!: string;
  @Output() handleClose: EventEmitter<any> = new EventEmitter();

  closeClick(e: Event): void {
    this.handleClose.emit(e);
  }
}
