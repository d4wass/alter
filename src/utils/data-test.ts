export function dataTest(selector: string) {
  return `[data-test="${selector}"]`;
}
