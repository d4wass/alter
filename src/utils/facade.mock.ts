import { Observable, BehaviorSubject } from 'rxjs';

export type FacadeMock<T> = {
  [P in keyof T]: T[P] extends Observable<infer U>
    ? BehaviorSubject<U>
    : T[P] extends (args: infer A) => Observable<infer R>
    ? (args: A) => BehaviorSubject<R>
    : T[P];
};
