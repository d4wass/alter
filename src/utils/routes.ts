export interface urlItem {
  description: string;
  url: string;
}

export const urlList: urlItem[] = [
  { description: 'Learn more', url: 'faq' },
  { description: 'Carculator', url: 'carculator' },
  { description: 'Become a host', url: 'host' },
  { description: 'Create Account', url: 'new-user' },
  { description: 'Your Cars', url: 'vehicles' }
];
