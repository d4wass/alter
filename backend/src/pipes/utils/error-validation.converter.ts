import { ValidationError } from 'class-validator';
import { ConvertedValidationError } from './validation-types';

export class ValidationErrorConverter {
  private _convertedErrors: Array<ConvertedValidationError> = [];
  public get convertedErrors(): Array<ConvertedValidationError> {
    return this._convertedErrors;
  }
  public set convertedErrors(value: Array<ConvertedValidationError>) {
    this._convertedErrors = value;
  }
  private _errors: ValidationError[];

  constructor(errors: ValidationError[]) {
    this._errors = errors;
    this.errorsConverter(this._errors);
  }

  errorsConverter(errors: ValidationError[]): void {
    errors.forEach((error) => {
      if (!error.children.length) {
        this._convertedErrors.push({ property: error.property, constraints: error.constraints });
      } else {
        this.errorsConverter(error.children);
      }
    });
  }
}
