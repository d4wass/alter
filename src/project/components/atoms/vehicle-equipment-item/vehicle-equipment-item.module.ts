import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { VehicleEquipmentItemComponent } from './vehicle-equipment-item.component';
import { MatListModule } from '@angular/material/list';
import { MatIconModule } from '@angular/material/icon';

@NgModule({
  declarations: [VehicleEquipmentItemComponent],
  imports: [CommonModule, MatListModule, MatIconModule],
  exports: [VehicleEquipmentItemComponent]
})
export class VehicleEquipmentItemModule {}
