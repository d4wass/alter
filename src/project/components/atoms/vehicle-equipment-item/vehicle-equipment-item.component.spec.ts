import { createComponentFactory, Spectator } from '@ngneat/spectator/jest';
import { dataTest } from '../../../../utils/data-test';
import { VehicleEquipmentItemComponent } from './vehicle-equipment-item.component';

describe('VehicleEquipmentItemComponent', () => {
  let component: VehicleEquipmentItemComponent;
  let spectator: Spectator<VehicleEquipmentItemComponent>;

  const createComponent = createComponentFactory({
    component: VehicleEquipmentItemComponent,
    shallow: true
  });

  beforeEach(() => {
    spectator = createComponent();
    component = spectator.component;
  });

  const selectors = {
    eqTitle: () => spectator.query(dataTest('title')),
    removeBtn: () => spectator.query(dataTest('remove-btn'))
  };

  it('should match to snapshot', () => {
    expect(spectator.fixture).toMatchSnapshot();
  });

  it('should create component', () => {
    expect(component).toBeTruthy();
  });

  describe('VehicleEquipmentItem', () => {
    it('should show equipment value', () => {
      const value = 'equipment';
      component.equipment = value;

      spectator.detectChanges();

      expect(component.equipment).toEqual(value);
      expect(selectors.eqTitle()).toHaveText(value);
    });

    it('should emit value on closeClick method call', () => {
      let e: any;
      const spy = jest.spyOn(component.handleClose, 'emit');
      component.closeClick(e);

      expect(spy).toHaveBeenCalledWith(e);
    });

    it('should emit value onn btn click', () => {
      const spy = jest.spyOn(component, 'closeClick');
      spectator.dispatchMouseEvent('.remove-btn', 'click');

      expect(spy).toHaveBeenCalled();
    });
  });
});
