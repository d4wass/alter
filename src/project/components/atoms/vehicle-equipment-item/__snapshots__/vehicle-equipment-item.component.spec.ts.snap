// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehicleEquipmentItemComponent should match to snapshot 1`] = `
<app-vehicle-equipment-item
  handleClose={[Function EventEmitter_]}
>
  <mat-list-item
    class="equipment-item"
  >
    <div
      data-test="title"
      matlistitemtitle=""
    >
    </div>
    <button
      class="remove-btn"
      data-test="remove-btn"
    >
      <mat-icon
        matlistitemicon=""
      >
        close
      </mat-icon>
    </button>
  </mat-list-item>
</app-vehicle-equipment-item>
`;
